src/rest-server/src/models/v2/job/k8s.js

```diff
@@ -177,16 +200,16 @@ const convertFrameworkDetail = async (framework) => {
       appLaunchedTime: new Date(framework.metadata.creationTimestamp).getTime(),
       appCompletedTime: new Date(framework.status.completionTime).getTime(),
       appExitCode: completionStatus ? completionStatus.code : null,
-      appExitSpec: {}, // TODO
-      appExitDiagnostics: completionStatus ? completionStatus.diagnostics : null,
-      appExitMessages: {
-        container: null,
-        runtime: null,
-        launcher: null,
-      },
-      appExitTriggerMessage: completionStatus ? completionStatus.diagnostics : null,
-      appExitTriggerTaskRoleName: null, // TODO
-      appExitTriggerTaskIndex: null, // TODO
+      appExitSpec: completionStatus ? generateExitSpec(completionStatus.code) : generateExitSpec(null),
+      appExitDiagnostics: diagInfo,
+      appExitMessages: diagInfo ? {
```

Comment:
>appExitMessages [](start = 6, length = 15)

This should contain the exit info after split the diagnostics to be launcher and runtime. 

And if runtime PAI_RUNTIME_ERROR_START cannot be found, the whole raw diagnostics should be just appExitMessages.launcher, and appExitMessages.runtime =null.
Otherwise, appExitMessages.launcher=null, appExitMessages.runtime =extracted.